h/static/scripts/annotator/adder.js

```diff
@@ -30,22 +27,88 @@ var ARROW_HEIGHT = 10;
 var ARROW_H_MARGIN = 20;
 
 /**
- * Controller for the 'adder' toolbar which appears next to the selection
+ * Create the DOM structure for the Adder.
+ *
+ * Returns the root DOM node for the adder, which may be in a shadow tree.
+ */
+function createAdderDOM(container) {
+  var element;
+
+  // If the browser supports Shadow DOM, use it to isolate the adder
+  // from the page's CSS
+  if (container.createShadowRoot) {
+    var root = container.createShadowRoot();
+    root.innerHTML = template;
+    element = root.querySelector('.js-adder');
+
+    // Load stylesheets required by adder into shadow DOM element
+    var adderStyles = Array.from(document.styleSheets).map(function (sheet) {
```

Comment:
Just had a realization that this is going to fail on sites with strict CSP policies that block inline styles. On a test site, the adder styling is indeed blocked, although the result is actually usable - you just get browser default styles for the buttons instead. Inline styles applied via JavaScript _do_ appear to work however because the adder is positioned correctly.

Perhaps another way to skin this cat would simply be to style all parts of the adder using inline styles set from JS, which are going to end up overriding page styles anyway.